### ISSUE TITLE
Container takes over an hour to build on my mbp 16GB ram. 

### ISSUE THREAD

**Issue by idvorkin** [semiosis/pen.el#32]
![image](https://user-images.githubusercontent.com/280981/147883846-b970c1c1-9809-4343-bd32-f8549b6e5796.png)

Even though it's a one time thing, it'll slow down folks who just want to kick the tires. I originally ran this on my remote AWS VM  and I lost my ability to ssh in, hopefully it'll resume once the build complete on there.

**Comment by mullikine** [comment]
Agreed. The docker container hasn't been diced up in to multiple tiers yet. I might add a warning about building for the time being. The docker build time is beyond belief, it can be compressed down to around 3.5 GB currently, which I had done a few weeks ago, but it grew again, as I haven't rebuilt the image in a while.

**Comment by weavermarquez** [comment]
I have to commend you for getting it in a container in the first place, because stands largely in the "Difficult [to containerize]" column as outlined in this fantastic guide:
[A Hacker’s Guide to Moving Linux Services into Containers](https://crunchtools.com/moving-linux-services-to-container)

I tried all day yesterday to get the image through podman in Fedora Silverblue 37, but could not get either to a usable state. The contained Emacs was able to execute, but ran into a number of runtime issues on boot and crashed. 

These incompatibility is fully understandable as the image is configured with a rooted docker in mind, even though I adjusted the setup scripts to operate without root and through podman. I also tried setting up a fresh VM of Fedora Workspace 37 without success, but I think that was mostly me.

Now, I don't know if this lines with your vision, so feel free to disagree:
- I think providing a Minimal Viable Product of pen.el as a simple emacs package with/without a simple external dependency would greatly help adoption.
- I'm not quite sure why simply interfacing with language models necessitates compiling emacs + integrating all these language toolchains.

To this end, I spent some time yesterday going through the dockerfile and build scripts, and would like to take a crack at a different containerization strategy, as outlined in the guide linked above. I'll have to get back to you next week with some actual progress, but please let us know what you think about pen.el's adoption strategy. After watching your demos on youtube, this seems like an important project and incredible tool for emacs, and that you've already done _most_ of the work to get to easy adoption.
